kettle/monitor_deployment.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: kettle-monitor
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: kettle-monitor
+    spec:
+      containers:
+      - name: kettle
+        image: gcr.io/k8s-gubernator/kettle:latest
+        imagePullPolicy: Always
+        env:
+        - name: GOOGLE_APPLICATION_CREDENTIALS
+          value: /etc/service-account/service-account.json
+        - name: MONITOR_PORT
+          value: "8000"
+        - name: MONITOR_TABLES
+          value: k8s-gubernator:build.all k8s-gubernator:build.week k8s-gubernator:build.day
+        volumeMounts:
+        - name: service
+          mountPath: /etc/service-account
+          readOnly: true
+      volumes:
+      - name: service
+        secret:
+          secretName: kettle-service-account
+---
+kind: Service
```

Comment:
What is the advantage of this over testgrid and a periodic prow job?

Reply:
Testgrid isn't the most appropriate tool for monitoring services. We should use service monitoring tools to monitor services.

Reply:
In particular: the kettle updater runs _continuously_ to stream build results, and wouldn't work well as a periodic prow job. Monitoring it by having a "test" that fails if a table was last modified too long ago would work, but it's weird.

Reply:
What alerting rules are you planning to write?

This would feel more true if we monitored the updater job, but that isn't what is happening.

Instead we're writing a new deployment and service that waits for http requests, at which point it fetches data from bigquery and returns it, which prometheus dumps into a time series database, which we can then write alerting rules for.

It seems like we can simplify this system by writing a job that reads bigquery directly and checks for problematic scenarios (lastUpdateTime is too old, number of results are too small). Then run this periodically on prow.

Failures would show up at https://k8s-testgrid.appspot.com/sig-testing-misc and we don't have anything new to manage.

Reply:
I can make this as an hourly periodic job that fails on stale results *and* pushes data to velodrome's influxdb instance (like the daily bigquery metrics job). Then we can use either alerting system, still get good historical metrics / graphing, and avoid an extra k8s deployment and service.

How does that sound?

Reply:
Fabulous! <3